### ISSUE TITLE
No editing of texts and no new shapes on new document

### ISSUE THREAD

**Issue by Olli** [prikhi/pencil#802]
Since update to Firefox 45.0 its not possible to edit texts or drag-n-drop new elements in an existing or new document. 
I assume this is caused by the update of Firefox.

**Comment by Olli** [comment]
Ok Update - this works ...in case you urgently need pencil for work.
firefox --app Outputs/Linux/application.ini

**Comment by itchybalaclava** [comment]
Olli, I had the same issue and got around it by simply downgrading Firefox. Please can you provide more detail about your suggested workaround? Thanks

**Comment by Olli** [comment]
git clone https://github.com/prikhi/pencil
cd pencil/build
./build.sh linux
cd Outputs/Linux
firefox -app application.ini 
Enough detail? :) Thats all I can provide.
Take a look at the description - it's provided here https://github.com/prikhi/pencil/ (just scroll down)

**Comment by prikhi** [comment]
Interesting, thanks for the report. Were no breaking changes in FF45 according to mozilla... Will investigate.

**Comment by devvv4ever** [comment]
I'm affected by this too... Can I somehow help debugging this?

**Comment by prikhi** [comment]
I'm pretty sure it's a bug in Firefox. There's nothing relevant in the release notes for FFv45.

I made a bug: https://bugzil.la/1259385

**Comment by prikhi** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1259385#c13

Gonna have to go through the nightlies & figure out which one breaks Pencil.

**Comment by Mannshoch** [comment]
I got only one significant change in the same Firefox Upgrade who break Pencil. Since this upgrade my Firefox contain his original Firefox Logo again. Until this tile I had the Pencil Logo instead of the Firefox Logo in the Running Firefox.

**Comment by dmiro** [comment]
I've the same problem. I cannot drag items to page.
Firefox 46.0.1

**Comment by Olli** [comment]
@prikhi any news on this issue?

**Comment by Mannshoch** [comment]
I installed Pencil native (not as addon) and it works.

**Comment by Olli** [comment]
Yes - if it's not an add-on it works as expected (I'd mention it here https://github.com/prikhi/pencil/issues/802#issuecomment-196254215 )

**Comment by piedgogo** [comment]
my ff 47.0 don't work too.

**Comment by piedgogo** [comment]
drag problem happens in my ff47.0 too.

**Comment by piedgogo** [comment]
build and run is not working too.

$ firefox -app application.ini
Error: Platform version '47.0' is not compatible with
minVersion >= 36.0
maxVersion <= 46.*

**Comment by DeepSilence** [comment]
Same problem with FF ESR 45.2.0, except for the shapes of my private collection _that are still drageable._
Are the default and private collections handled differently by pencil?

**Comment by prikhi** [comment]
I don't really have the time to figure this out, sorry guys. If you're getting this bug, I suggest moving on to http://github.com/evolus/pencil.

I'll keep bumping maxVersion hoping it'll be fixed in the future but if you want to continue using v2 either use an older firefox, or use xulrunner/palemoon. I'm using this repo, mostly with xulrunner but also with palemoon.

**Comment by gabrielrincon** [comment]
Confirm on firefox 50.0. Dont work, i cant drag and drop the shapes on the document.

**Comment by lofidevops** [comment]
@prikhi Thanks so much for all your maintenance effort, it really makes a difference. Based on your feedback on this ticket, I downloaded `Pencil-2.0.21-win32.installer.exe` from the releases page and it works perfectly. I'm now using it rather than the XPI, and eagerly watching progress on Pencil 3.0 :)

**Comment by umiphos** [comment]
[base answer](http://askubuntu.com/questions/125980/how-do-i-install-xulrunner)

If you are currently on Ubuntu 16.04 maybe the default mozilla will not allow you to run this, but you can always run it with xulrunner.
[Ollie teach us how build from source](https://github.com/prikhi/pencil/issues/802#issuecomment-196254215)

FFVER=`grep -Po  "\d{2}\.\d+" /usr/lib/firefox/platform.ini`
ARCH=`uname -p`
XURL=https://ftp.mozilla.org/pub/mozilla.org/xulrunner/releases/$FFVER/runtimes/xulrunner-$FFVER.en-US.linux-$ARCH.tar.bz2
cd /opt
sudo sh -c "wget -O- $XURL | tar -xj"
sudo ln -s /opt/xulrunner/xulrunner /usr/bin/xulrunner
sudo ln -s /opt/xulrunner/xpcshell /usr/bin/xpcshell

[If the package fails use this on XURL=](http://ftp.mozilla.org/pub/xulrunner/releases/41.0b9/runtimes/xulrunner-41.0b9.en-US.linux-x86_64.tar.bz2)

**Comment by paulewetzel** [comment]
Does not work for me either on Ubuntu 16.04. Starts but cannot drag elements.
